addons/stock_account/migrations/8.0.1.1/post-migrate.py

```diff
@@ -42,7 +44,65 @@ def create_properties(cr, pool):
                                context=ctx)
 
 
-@openupgrade.migrate()
+def propagate_invoice_state(cr):
+    """ Invoice state is now propagated from sale order to procurement to
+    stock move and picking. We trace it back from the picking and update
+    stock moves and procurements. """
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE stock_move sm
```

Comment:
That was my response when I got this query from my colleague @NL66278. We discussed and found out using the postgresql function 'explain analyse' that this subquery variant was 10 to 20 times as fast as the 'update...from' statement.

I know I have been going round myself requesting in reviews that subqueries be eliminated based on the assumption that joins are usually quicker. Turns out I was wrong.

I do agree that the join variant is more readable, and I find these queries easier to write, which is the reason that the other queries in this PR are still join based. None of these particular queries formed a bottleneck in the migrations that we performed. But it seemed silly not to use a more performant query if it is available.